crates/biome_css_analyze/src/lint/nursery/no_duplicate_selectors.rs

```diff
@@ -0,0 +1,327 @@
+                        } else {
+                            child
+                                .clone()
+                                .cast::<AnyCssRelativeSelector>()
+                                .unwrap()
+                                .text()
+                        };
+
+                        if let Some(first) = this_list_resolved_list.get(&selector_text) {
+                            output.push(DuplicateSelector {
+                                first: first.selector_node.clone(),
+                                duplicate: child.clone(),
+                            });
+                            return None;
+                        }
+
+                        this_list_resolved_list.insert(ResolvedSelector {
+                            selector_text: selector_text.clone(),
+                            selector_node: child,
+                        });
+                        Some(selector_text)
+                    })
+                    .collect();
+                selector_list_mapped.sort();
+
+                for r in resolve_nested_selectors(selector_list_mapped.join(","), rule) {
+                    let split: Vec<&str> = r.split_whitespace().collect();
+                    let normalized = split.join(" ").to_lowercase();
+                    if let Some(first) = resolved_list.get(&normalized) {
+                        output.push(DuplicateSelector {
+                            first: first.selector_node.clone(),
+                            duplicate: selector_list.clone(),
+                        });
+                    } else {
+                        resolved_list.insert(ResolvedSelector {
+                            selector_text: normalized.clone(),
+                            selector_node: selector_list.clone().into(),
+                        });
+                    }
+                }
+            }
+        }
+        output
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, node: &Self::State) -> Option<RuleDiagnostic> {
+        //
+        // Read our guidelines to write great diagnostics:
+        // https://docs.rs/biome_analyze/latest/biome_analyze/#what-a-rule-should-say-to-the-user
+        //
+        let duplicate_text = node.duplicate.to_string();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.duplicate.text_trimmed_range(),
+                markup! {
+                    "Duplicate selectors may result in unintentionally overriding rules: "<Emphasis>{ duplicate_text }</Emphasis>
+                },
+            )
+            .detail(node.first.text_trimmed_range(), "Please consider moving the rule's contents to the first occurence:")
+            .note(markup! {
+                "Remove duplicate selectors within the rule"
+            }),
+        )
+    }
+}
+
+fn resolve_nested_selectors(selector: String, this_rule: CssSyntaxNode) -> Vec<String> {
```

Comment:
Can we have some documentation that explains this function? It's quite long, so I suppose the logic needs some explanation